zerver/lib/actions.py

```diff
@@ -803,6 +804,7 @@ def do_change_user_email(user_profile: UserProfile, new_email: str) -> None:
 def do_start_email_change_process(user_profile: UserProfile, new_email: str) -> None:
     old_email = user_profile.email
     user_profile.email = new_email
+    user_profile.actual_email = new_email
```

Comment:
I think this change is wrong.  I see why you did it, but I'm pretty sure it's actually a (mostly harmless) bug that we have `user_profile.email = new_email` on the previous line; we don't call `user_profile.save()` and thus it's very likely that line has no effect.  Can you add a commit at the start of this branch which removes the line before this, manually test that the email change workflow still works, and then amend this commit to just not edit `do_start_email_change_process`?